resources/integration_test/testdata/integration.yml

```
@@ -91,18 +91,18 @@ dataset:
     docsExtJson:
       - {
         _id: 0,
+        array: [1,2,3,"$oid": "000000000000000000000003", "$timestamp": {"t":200, "i":0}],
```

I was a little surprised to see that this parsed at all -- does YAML let you specify a document as key-value pairs without curly braces even inside a bracketed array?

Either way, I'd suggest adding the curly braces just to make it a bit more explicit
```suggestion
        array: [1,2,3,{"$oid": "000000000000000000000003"}, {"$timestamp": {"t":200, "i":0}}],
```

---

Added the curly braces. 

I checked to see what the row looks like after SnakeYaml loads it and I see that it adds the curly braces, which is probably why the parsing worked:
in yml file:
array: [1,2,3,"$oid": "000000000000000000000003", "$timestamp": {"t":200, "i":0}],
println in Java of the entry:
array=[1, 2, 3, {$oid=000000000000000000000003}, {$timestamp={t=200, i=0}}]